docs/release_notes/v1.9.0.md

```diff
@@ -0,0 +1,994 @@
+```
+
+For more details, refer to the [Search Documentation](https://spiceai.org/docs/features/search) and [Embeddings Documentation](https://spiceai.org/docs/features/embeddings).
+
+### Dedicated Query Thread Pool (Now Enabled by Default)
+
+**Dedicated Query Thread Pool**: Query execution and accelerated refreshes now run on their own dedicated thread pool, separate from the HTTP server. This prevents heavy query workloads from slowing down API responses, keeping health checks fast and avoiding unnecessary Kubernetes pod restarts under load.
+
+This feature was opt-in in previous releases and is now **enabled by default**. To disable it and revert to the previous behavior, add the following spicepod.yaml configuration:
+
+```yaml
+runtime:
+  params:
+    dedicated_thread_pool: none
+```
+
+For more details, refer to the [Runtime Configuration Documentation](https://spiceai.org/docs/reference/spicepod/runtime).
+
+### Query Performance Optimizations
+
+**Stale-While-Revalidate Cache Control**: Query results now support "stale-while-revalidate" cache control, allowing stale cached data to be served immediately while asynchronously refreshing the cache entry in the background. This improves response times for frequently-accessed queries while maintaining data freshness. Requires cache key type to be set to "sql (raw)" for proper operation.
+
+**Optimized Prepared Statements**: Prepared statement handling has been optimized for better performance with parameterized queries, reducing planning overhead and improving execution time for repeated queries.
+
+**Large RecordBatch Chunking**: Large Arrow `RecordBatch` objects are now automatically chunked to control memory usage during query execution, preventing memory exhaustion for queries returning large result sets.
+
+### Query Result Caching: Stale-While-Revalidate
+
+**HTTP Cache-Control Support**: The query result cache now supports the `stale-while-revalidate` [Cache-Control directive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#stale-while-revalidate), enabling faster response times by serving stale cached results immediately while asynchronously refreshing the cache in the background. This feature is particularly useful for applications that can tolerate slightly stale data in exchange for improved performance.
+
+**Zstd Compression Encoding**: Query result caching now supports optional [Zstandard (zstd)](https://facebook.github.io/zstd/) compression encoding to reduce memory usage for cached query results. This is particularly beneficial for large result sets, reducing cache memory footprint while maintaining fast decompression times. Encoding can be configured via the `encoding` parameter with options `none` (default) or `zstd`.
+
+Example configuration:
+
+```yaml
+runtime:
+  results_cache:
+    enabled: true
+    cache_max_size: 128MiB
+    item_ttl: 1m
+    encoding: zstd # Enable zstd compression
+```
+
+**How it works**:
```

Comment:
This reads to me as the "How it works" for the zstd compression, we should probably move above to right after the `HTTP Cache-Control Support` section